packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -538,24 +522,114 @@ export function checkVerifyClaimsForConfig(
       }
     }
   }
+  // TODO(POD-P2): List membership claims
+}
+
+/**
+ * Picks the smallest available circuit in this family which can handle the
+ * size parameters of a desired configuration.
+ *
+ * @param circuitReq the circuit size requirements
+ * @returns the circuit description, or undefined if no circuit can handle
+ *   the required parameters.
+ * @throws Error if there are no circuits satisfying the given requirements.
+ */
+export function circuitDescForRequirements(
+  circuitReq: GPCRequirements
+): ProtoPODGPCCircuitDesc {
+  for (const circuitDesc of ProtoPODGPC.CIRCUIT_FAMILY) {
+    if (circuitDescMeetsRequirements(circuitDesc, circuitReq)) {
+      return circuitDesc;
+    }
+  }
+
+  throw new Error(
+    `There are no circuit parameters satisfying these requirements: ${circuitReq}`
+  );
+}
+
+/**
+ * Checks whether a described circuit can meet given GPC size requirements.
+ * This will be true if each of the circuit's parameters is greater than or
+ * equal to the required value.
+ *
+ * @param circuitDesc description of the circuit to check
+ * @param circuitReq the circuit size requirements
+ * @returns `true` if the circuit meets the requirements.
+ */
+export function circuitDescMeetsRequirements(
+  circuitDesc: ProtoPODGPCCircuitDesc,
+  circuitReq: GPCRequirements
+): boolean {
+  // Check tuple parameter compatibility.
+  // Indicate negative tuple requirement check if `requiredNumTuples` throws due
+  // to an invalid choice of parameters (i.e. `params.tupleArity < 2`).
+  const tupleCheck = ((): boolean => {
+    try {
+      // The circuit description should have enough tuples of arity `tupleArity` to
+      // cover all input tuples.
+      return (
+        circuitDesc.maxTuples >=
+        circuitReq.tupleArities
```

Comment:
The algorithm here is worth a comment.  I.e. that requiredNumTuples determines how many circuit tuples are required to represent one input tuple, and that the uses need to be independent (can't overlap) so the net result is a sum.